### PR TITLE
Add `BorderType` enum to bmsmsd

### PR DESCRIPTION
**Pull request by dyceron**
Adds an enum to `BorderType` in the bmsmsd. It's kinda weird and a lot, but it covers all of the tile borders.

### CODE REVIEW

**Review by duncathan — CHANGES_REQUESTED** [review]
this should be a `FlagsEnum` with just 8 entries (4 normal edges, 4 hidden edges)

**Comment by dyceron** [comment]
> this should be a `FlagsEnum` with just 8 entries (4 normal edges, 4 hidden edges)

You mean instead of the big enum? Idk what that is.

**Comment by dyceron** [comment]
I sorta understand but what to map the enum values to is confusing

**Comment by duncathan** [comment]
the values you have for 1, 2, 4, 8, 16, 32, 64, and 128

File: src/mercury_engine_data_structures/formats/bmsmsd.py [hunk 1]
```diff
@@ -45,7 +115,7 @@
                 "bottom_left" / CVector2D,
                 "top_right" / CVector2D,
             ),
-            "border_type" / Int32sl,
+            "border_type" / BorderType,
             "tile_type" / TileType,
```
Comment (duncathan) [inline src/mercury_engine_data_structures/formats/bmsmsd.py:119]:
```py
"border_type" / FlagsEnum(Int32sl,
    TOP=1,
    BOTTOM=2,
    LEFT=4,
    RIGHT=8,
    OPEN_TOP=16,
    OPEN_BOTTOM=32,
    OPEN_LEFT=64,
    OPEN_RIGHT=128,
)
```

**Review by duncathan — APPROVED** [review]

**Pull request merged**